### ISSUE TITLE
plugin disabled after VM reset

### ISSUE THREAD

**Issue by LelandSindt** [moby/moby#36639]
**Description**

Plugins install, and work until after the VM is reset* at which point it displays enabled: false and attempting to enable it reports
Error response from daemon: dial unix /run/docker/plugins/[conatiner ID]/rexray.sock: connect: no such file or directory

*after first seeing this behaviour last week, and spending the morning testing it.. the failure is  consistent after the VM is reset from the GCE console.

**Steps to reproduce the issue:**
--Create a new GCE VM 
`# gcloud beta compute --project "YOUR PROJECT" instances create "rexray" --zone "us-central1-b" --machine-type "n1-standard-1" --subnet "default" --maintenance-policy "MIGRATE" --service-account "YOUR ACCOUNT" --scopes "https://www.googleapis.com/auth/compute","https://www.googleapis.com/auth/servicecontrol","https://www.googleapis.com/auth/service.management.readonly","https://www.googleapis.com/auth/logging.write","https://www.googleapis.com/auth/monitoring.write","https://www.googleapis.com/auth/trace.append","https://www.googleapis.com/auth/devstorage.read_write" --min-cpu-platform "Automatic" --image "ubuntu-1604-xenial-v20180306" --image-project "ubuntu-os-cloud" --boot-disk-size "10" --boot-disk-type "pd-standard" --boot-disk-device-name "rexray"`

-- SSH to the newly created VM.
`# ssh IP.ADDRESS.OF.VM`

-- Install Docker (add yourself to the docker group.)
`# curl -fsSL get.docker.com -o /tmp/get-docker.sh; sh /tmp/get-docker.sh; sudo usermod -aG docker $(whoami)`

-- Install rexray/gcepd plugin.
```
# sudo docker plugin install rexray/gcepd:latest GCEPD_TAG=rexray GCEPD_CONVERTUNDERSCORES=true --grant-all-permissions
latest: Pulling from rexray/gcepd
8ba31ee8b04f: Download complete 
Digest: sha256:ed47c39cc6ced342aeefe776bdaee9974152e91946259167a82d942e0a36edcf
Status: Downloaded newer image for rexray/gcepd:latest
Installed plugin rexray/gcepd:latest
```

-- prove the plugin is working... 
```
# sudo docker plugin ls
ID                  NAME                  DESCRIPTION                        ENABLED
d1a9fc4d0f67        rexray/gcepd:latest   REX-Ray for GCE Persistent Disks   true
# sudo docker volume create --driver rexray/gcepd:latest rexray-disk
rexray_disk
# sudo docker volume ls
DRIVER                VOLUME NAME
rexray/gcepd:latest   rexray-disk
# docker container run -it --rm --volume rexray-disk:/data alpine /bin/sh
Unable to find image 'alpine:latest' locally
latest: Pulling from library/alpine
ff3a5c916c92: Pull complete 
Digest: sha256:7b848083f93822dd21b0a2f14a110bd99f6efb4b838d499df6d04a49d0debf8b
Status: Downloaded newer image for alpine:latest
/ # echo "hello" > /data/world
/ # cat /data/world
hello
/ # exit

```
-- reboot... 
`# sudo reboot`
.... wait

-- SSH to the rebooted VM.
`# ssh IP.ADDRESS.OF.VM
`
-- Prove the plugin is still working... 
```
docker plugin ls
ID                  NAME                  DESCRIPTION                        ENABLED
d1a9fc4d0f67        rexray/gcepd:latest   REX-Ray for GCE Persistent Disks   true
# docker volume ls
DRIVER                VOLUME NAME
rexray/gcepd:latest   rexray-disk
# sudo docker container run -it --rm --volume rexray-disk:/data alpine /bin/sh
/ # cat /data/world
hello
/ # exit

```
-- Reset the VM
```
# gcloud compute instances reset rexray
Updated [https://www.googleapis.com/compute/v1/projects/YOUR PROJECT/zones/us-central1-b/instances/rexray].
```
...wait

-- SSH to the reset VM.
`# ssh IP.ADDRESS.OF.VM
`

-- Prove the plugin is disabled... 
```
# docker plugin ls
ID                  NAME                  DESCRIPTION                        ENABLED
6965f00d072b        rexray/gcepd:latest   REX-Ray for GCE Persistent Disks   false
# docker plugin enable rexray/gcepd:latest 
Error response from daemon: dial unix /run/docker/plugins/6965f00d072b70808b3984ac55568f6d917ec32b9c568259982cf1b90ccea1f5/rexray.sock: connect: no such file or directory

# journalctl -u docker.service --no-tail -o cat |grep plugin
...
time="2018-03-19T19:44:48.921452325Z" level=error msg="Sending SIGTERM to plugin failed with error: no such container"
time="2018-03-19T19:44:48.922224469Z" level=error msg="failed to enable plugin 'rexray/gcepd:latest': dial unix /run/docker/plugins/6965f00d072b70808b3984ac55568f6d917ec32b9c568259982cf1b90ccea1f5/rexray.sock: connect: no such file or directory"
time="2018-03-19T19:44:49.205936881Z" level=warning msg="Error restoring volume" driver="rexray/gcepd:latest" error="error while checking if volume \"rexray-disk\" exists in driver \"rexray/gcepd:latest\": error looking up volume plugin rexray/gcepd:latest: plugin rexray/gcepd:latest found but disabled" volume=rexray-disk
time="2018-03-19T19:46:49.101536207Z" level=error msg="Sending SIGTERM to plugin failed with error: no such container"
time="2018-03-19T19:46:49.102283851Z" level=error msg="Handler for POST /v1.36/plugins/rexray/gcepd:latest/enable returned error: dial unix /run/docker/plugins/6965f00d072b70808b3984ac55568f6d917ec32b9c568259982cf1b90ccea1f5/rexray.sock: connect: no such file or directory"
...

```
At this point with no combination of resets/reboots have I been able to get the plugin to load/enable... However, removing and installing the plugin will bring everything back to working again. 

```
# docker plugin rm rexray/gcepd:latest 
rexray/gcepd:latest
leland@rexray:~$ docker volume ls
DRIVER              VOLUME NAME
# sudo docker plugin install rexray/gcepd:latest GCEPD_TAG=rexray GCEPD_CONVERTUNDERSCORES=true --grant-all-permissions
latest: Pulling from rexray/gcepd
8ba31ee8b04f: Download complete 
Digest: sha256:ed47c39cc6ced342aeefe776bdaee9974152e91946259167a82d942e0a36edcf
Status: Downloaded newer image for rexray/gcepd:latest
Installed plugin rexray/gcepd:latest
# docker plugin ls
ID                  NAME                  DESCRIPTION                        ENABLED
086139fb6a05        rexray/gcepd:latest   REX-Ray for GCE Persistent Disks   true
# docker volume ls
DRIVER                VOLUME NAME
rexray/gcepd:latest   rexray-disk
```

**Describe the results you received:**
The plugin is disabled after VM reset.

**Describe the results you expected:**
The plugin would be enabled after VM reset. 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:       18.02.0-ce
 API version:   1.36
 Go version:    go1.9.3
 Git commit:    fc4de44
 Built: Wed Feb  7 21:16:33 2018
 OS/Arch:       linux/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.02.0-ce
  API version:  1.36 (minimum version 1.12)
  Go version:   go1.9.3
  Git commit:   fc4de44
  Built:        Wed Feb  7 21:15:05 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
docker version
Client:
 Version:       18.02.0-ce
 API version:   1.36
 Go version:    go1.9.3
 Git commit:    fc4de44
 Built: Wed Feb  7 21:16:33 2018
 OS/Arch:       linux/amd64
 Experimental:  false
 Orchestrator:  swarm

Server:
 Engine:
  Version:      18.02.0-ce
  API version:  1.36 (minimum version 1.12)
  Go version:   go1.9.3
  Git commit:   fc4de44
  Built:        Wed Feb  7 21:15:05 2018
  OS/Arch:      linux/amd64
  Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: No swap limit support
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

As shown in the steps to reproduce this is a GCE VM.

I mistakenly reported this to the rexray team... https://github.com/rexray/rexray/issues/1191 then realized it was not rexray specific with I used the vieux/sshfs plugin

```
lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.4 LTS
Release:        16.04
Codename:       xenial
```

**Comment by LelandSindt** [comment]
After downgrading to 18.01.0-ce I no longer see this behaviour...

**Comment by anusha-ragunathan** [comment]
I tried to reproduce the issue on 18.02-ce with `vieux/sshfs` on AWS (Dont have access to GCE) and I dont see it. A reset on AWS is done using `sudo reboot`. Plugins's state is persisted across resets, as expected.

Analyzing dockerd logs from https://github.com/rexray/rexray/issues/1191
- `reload()` calls to plugin creation fails. I'm not sure why. Need to dig deeper.
- `pluginPostStart` is incorrectly called on this error condition. So a new client is created for a non-existent plugin, which errors out expectedly and an attempt is made to shutdown this non-existent plugin.   https://github.com/docker/docker-ce/blob/18.02/components/engine/plugin/manager_linux.go#L65-L85

```
time="2018-03-19T20:02:50Z" level=info msg="containerd successfully booted in 0.028863s" module=containerd
time="2018-03-19T20:03:03.189828523Z" level=error msg="Sending SIGTERM to plugin failed with error: no such container"
time="2018-03-19T20:03:03.190658138Z" level=error msg="failed to enable plugin 'rexray/gcepd:latest': dial unix /run/docker/plugins/086139fb6a05181557cea3148859654978fbd6ff511474a0cfbe2a8128d55c1f/rexray.sock: connect: no such file or directory"
```

**Comment by anusha-ragunathan** [comment]
@LelandSindt : Can you try running a container with restart policy set to `always`, on GCE, with 18.02-ce docker-engine and reset the VM? I'm trying to see if this is a generic issue across containers and plugins.

Something like `docker run -d --restart=always busybox top` and reset the VM. Observe if the container comes up after the reset, as expected.

**Comment by LelandSindt** [comment]
@anusha-ragunathan 

 `sudo reboot` will not trigger the issue, it requires a reset. 

... think clean reboot vs pulling the power/restoring power. In CGE a reset can be preformed using the `gcloud compute instances reset <instance>` command

As for you request to test with a `--restart always` container, it is behaving similar to the plugin, yes.... 

```
leland@rexray:~$ docker run -d --restart=always busybox top
Unable to find image 'busybox:latest' locally
latest: Pulling from library/busybox
d070b8ef96fc: Pull complete 
Digest: sha256:2107a35b58593c58ec5f4e8f2c4a70d195321078aebfadfbfb223a2ff4a4ed21
Status: Downloaded newer image for busybox:latest
5097a145295421f9ef27b2ed88dfef93be474cbc6ae5e42f2378d81b4f145bb9
leland@rexray:~$ docker container ls
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
5097a1452954        busybox             "top"               10 seconds ago      Up 10 seconds                           nifty_mayer
leland@rexray:~$ sudo reboot
leland@x1:~$ ssh rexray
leland@rexray:~$ docker container ls --all
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS               NAMES
5097a1452954        busybox             "top"               About a minute ago   Up 8 seconds                            nifty_mayer
leland@rexray:~$ exit
leland@x1:~$ gcloud compute instances reset rexray                                                                                                                                        
Updated [https://www.googleapis.com/compute/v1/projects/rexray-198519/zones/us-central1-b/instances/rexray].
leland@x1:~$ ssh rexray
leland@rexray:~$ docker container ls --all
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                        PORTS               NAMES
5097a1452954        busybox             "top"               2 minutes ago       Exited (255) 27 seconds ago                       nifty_mayer
```

**Comment by LelandSindt** [comment]
I have not been able to find an equivalent EC2 command to the GCE reset command I listed... 

The closest I have been able to find is this....  
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-reboot.html
"... we perform a hard reboot if the instance does not cleanly shut down within four minutes. "

You would have to intentionally hang the shut down process forcing EC2 to "preform a hard reboot"

**Comment by LelandSindt** [comment]
Additional logs with context... 

```
leland@rexray:~$ docker container run -d --restart=always busybox top                                                                             
dc4fd777840963e85c0ba07934c4485429eb8ba2cc597090bdf6a3977ed624cb
leland@rexray:~$ docker container ls
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
dc4fd7778409        busybox             "top"               6 seconds ago       Up 5 seconds                            sleepy_bose
leland@rexray:~$ exit
leland@x1:~$ gcloud compute instances reset rexray
Updated [https://www.googleapis.com/compute/v1/projects/rexray-198519/zones/us-central1-b/instances/rexray].
leland@x1:~$ ssh rexray
leland@rexray:~$ docker container ls --all
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                        PORTS               NAMES
dc4fd7778409        busybox             "top"               2 minutes ago       Exited (255) 42 seconds ago                       sleepy_bose
leland@rexray:~$ sudo journalctl -o cat -u docker.service
Starting Docker Application Container Engine...
time="2018-03-21T12:37:08.214518715Z" level=info msg="libcontainerd: started new docker-containerd process" pid=1422
time="2018-03-21T12:37:08Z" level=info msg="starting containerd" module=containerd revision=9b55aab90508bd389d7654c4baf173a981477d55 version=v1.0.
1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.content.v1.content"..." module=containerd type=io.containerd.content.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.snapshotter.v1.btrfs"..." module=containerd type=io.containerd.snapshott
er.v1
time="2018-03-21T12:37:08Z" level=warning msg="failed to load plugin io.containerd.snapshotter.v1.btrfs" error="path /var/lib/docker/containerd/da
emon/io.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter" module=containerd
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.snapshotter.v1.overlayfs"..." module=containerd type=io.containerd.snaps
hotter.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.metadata.v1.bolt"..." module=containerd type=io.containerd.metadata.v1
time="2018-03-21T12:37:08Z" level=warning msg="could not use snapshotter btrfs in metadata plugin" error="path /var/lib/docker/containerd/daemon/i
o.containerd.snapshotter.v1.btrfs must be a btrfs filesystem to be used with the btrfs snapshotter" module="containerd/io.containerd.metadata.v1.b
olt"
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.differ.v1.walking"..." module=containerd type=io.containerd.differ.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.gc.v1.scheduler"..." module=containerd type=io.containerd.gc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.containers"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.content"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.diff"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.events"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.healthcheck"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.images"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.leases"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.namespaces"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.snapshots"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.monitor.v1.cgroups"..." module=containerd type=io.containerd.monitor.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.runtime.v1.linux"..." module=containerd type=io.containerd.runtime.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.tasks"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.version"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg="loading plugin "io.containerd.grpc.v1.introspection"..." module=containerd type=io.containerd.grpc.v1
time="2018-03-21T12:37:08Z" level=info msg=serving... address="/var/run/docker/containerd/docker-containerd-debug.sock" module="containerd/debug"
time="2018-03-21T12:37:08Z" level=info msg=serving... address="/var/run/docker/containerd/docker-containerd.sock" module="containerd/grpc"
time="2018-03-21T12:37:08Z" level=info msg="containerd successfully booted in 0.025160s" module=containerd
time="2018-03-21T12:37:08.659042556Z" level=info msg="[graphdriver] using prior storage driver: overlay2"
time="2018-03-21T12:37:08.812533111Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
time="2018-03-21T12:37:08.813111308Z" level=warning msg="Your kernel does not support swap memory limit"
time="2018-03-21T12:37:08.813410233Z" level=warning msg="Your kernel does not support cgroup rt period"
time="2018-03-21T12:37:08.813563687Z" level=warning msg="Your kernel does not support cgroup rt runtime"
time="2018-03-21T12:37:08.814357765Z" level=info msg="Loading containers: start."
time="2018-03-21T12:37:08.838121344Z" level=error msg="Failed to load container mount 281e907605b62378133ae95510901955a235ad6f2f0abe4b4bd98c476e972df2: mount does not exist"
time="2018-03-21T12:37:08.843766085Z" level=error msg="Failed to load container mount f3a72837cf315c0f21022d1b3ebc8f898494c1367adbebae3121156090259a3f: mount does not exist"
time="2018-03-21T12:37:08.864534731Z" level=error msg="dc4fd777840963e85c0ba07934c4485429eb8ba2cc597090bdf6a3977ed624cb cleanup: failed to delete container from containerd: no such container"
time="2018-03-21T12:37:09.212655907Z" level=info msg="Removing stale sandbox 087ecd2af6099a2b4ae9b4b094b205298fb5240674ea0438844a7ce0150be386 (dc4fd777840963e85c0ba07934c4485429eb8ba2cc597090bdf6a3977ed624cb)"
time="2018-03-21T12:37:09.220059027Z" level=warning msg="Error (Unable to complete atomic operation, key modified) deleting object [endpoint 54fcef7ebb0362119985e2ae491456a9bc8a34c423c2b9cfc2ad3e5bbec493bf ddae4171b67cfb8b95c4dfc0cefbdcf0fffc9a9129b2de0dbba1de1f8914f2cc], retrying...."
time="2018-03-21T12:37:09.259991905Z" level=info msg="Default bridge (docker0) is assigned with an IP address 172.17.0.0/16. Daemon option --bip can be used to set a preferred IP address"
time="2018-03-21T12:37:09.424260378Z" level=error msg="dc4fd777840963e85c0ba07934c4485429eb8ba2cc597090bdf6a3977ed624cb cleanup: failed to delete container from containerd: no such container"
time="2018-03-21T12:37:09.425755489Z" level=error msg="Failed to start container dc4fd777840963e85c0ba07934c4485429eb8ba2cc597090bdf6a3977ed624cb: container \"dc4fd777840963e85c0ba07934c4485429eb8ba2cc597090bdf6a3977ed624cb\": already exists"
time="2018-03-21T12:37:09.426125700Z" level=info msg="Loading containers: done."
time="2018-03-21T12:37:09.798769134Z" level=info msg="Docker daemon" commit=fc4de44 graphdriver(s)=overlay2 version=18.02.0-ce
time="2018-03-21T12:37:09.800742501Z" level=info msg="Daemon has completed initialization"
Started Docker Application Container Engine.
time="2018-03-21T12:37:09.814270484Z" level=info msg="API listen on /var/run/docker.sock"
```

**Comment by LelandSindt** [comment]
sometimes... ?depending on my timing? after a reset `docker container ls --all`  returns an empty list rather than showing the failed container.

**Comment by LelandSindt** [comment]
one last note/observation for now... (I need to get some work done today ;) ) 

The list of "Failed to load container mount" errors grows with every reset/cycle... 

```
time="2018-03-21T12:56:05.148742086Z" level=info msg="Loading containers: start."
time="2018-03-21T12:56:05.153092497Z" level=error msg="Failed to load container mount 2397dc4cb2e1da9e638577f1fe3c4fb8f1f3a03a322d490751a6ea823b07bfe9: mount does not exist"
time="2018-03-21T12:56:05.155836917Z" level=error msg="Failed to load container mount 281e907605b62378133ae95510901955a235ad6f2f0abe4b4bd98c476e972df2: mount does not exist"
time="2018-03-21T12:56:05.158966327Z" level=error msg="Failed to load container mount 33794c56cc047fce9655903c1dbee74229b683804fe1b2b8c87ff1b88255a01d: mount does not exist"
time="2018-03-21T12:56:05.162068526Z" level=error msg="Failed to load container mount bbcfedb5f6b555cf3511ab8e93469e0d0a76798e54ea52b8f3103c759a9ba389: mount does not exist"
time="2018-03-21T12:56:05.187550136Z" level=error msg="Failed to load container mount f3a72837cf315c0f21022d1b3ebc8f898494c1367adbebae3121156090259a3f: mount does not exist"
time="2018-03-21T12:56:05.609565283Z" level=info msg="Removing stale sandbox a27879a738ed18e524bb08e9c1858f3038b2686f64d0ee6b1056cc63d3a0c525 (bbcfedb5f6b555cf3511ab8e93469e0d0a76798e54ea52b8f3103c759a9ba389)"
```

**Comment by anusha-ragunathan** [comment]
@LelandSindt : What's the output of `docker info`? (not `docker version`)

**Comment by anusha-ragunathan** [comment]
Looking at the commit diffs between 18.01 (pass) and 18.02 (fail), there's one commit related to mounts (https://github.com/docker/docker-ce/commit/a19065e95147f834204fc1492343e6c3a96a8c4b) and there's a new containerd vendoring (https://github.com/docker/docker-ce/commit/c6678dac698e3eed3f79af1522ec87802bc2414a).

Those would be my first suspects. cc @cpuguy83 @thaJeztah 

Having said that, we really need a GCE repro before spending more time on this.

**Comment by thaJeztah** [comment]
Probably silly thought, but  given that it’s only after a reboot; is all storage in `/var/lib/docker` on the same mount, or is an extra disk attached to the VM for storage?

**Comment by LelandSindt** [comment]
@anusha-ragunathan  I have updated the first post to with the `docker info` output.. (I had mistakenly pasted docker version twice, sorry)

@thaJeztah  good question... however, there is only one mount/volume/disk involved... 

```
leland@rexray:~$ df -h
Filesystem      Size  Used Avail Use% Mounted on
udev            1.8G     0  1.8G   0% /dev
tmpfs           370M  5.2M  365M   2% /run
/dev/sda1       9.7G  1.5G  8.2G  16% /
tmpfs           1.9G     0  1.9G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           1.9G     0  1.9G   0% /sys/fs/cgroup
tmpfs           370M     0  370M   0% /run/user/1001
```

**Comment by LelandSindt** [comment]
I would be glad to work with ya'll (via webex/zoom/google hangouts) to demo/reproduce this in our GCE Env...

**Comment by LelandSindt** [comment]
18.03.0-ce seems to have resolved the issue... 

both `--restart always` containers and plugins are "surviving" a "hard reset"

However, the `Failed to load container mount 2397dc4cb2e1da9e638577f1fe3c4fb8f1f3a03a322d490751a6ea823b07bfe9: mount does not exist` error messages persist... 

Please let me know if you would like any further details/logs/etc.

**Comment by anusha-ragunathan** [comment]
18.03-ce has another vendor of containerd. I'm assuming that fixed whatever got temporarily broke in 18.02-ce. Given that, I would consider the issue resolved.

@LelandSindt : For the `Failed to load container mount` issue, I would recommend opening a new issue.
